### PR TITLE
[VPU] removal deprecated test

### PR DESCRIPTION
**Pull request by KorA6**
### Details:
 - Switching the `ROIFeatureExtractor` test from the old test infrastructure to the new one
 - Changes in `ov_subgraph.cpp` and `ov_behavior_test_utils.hpp` is a workaround for "Failed to allocate graph: MYRIAD device is not opened." issue

### Tickets:
 - -72380
 - -80504

### CODE REVIEW

**Review by DariaMityagina — APPROVED** [review]

**Review by AntonDudchenko — APPROVED** [review]

**Comment by KorA6** [comment]
@iefode   
Take a look please

**Comment by KorA6** [comment]
@iefode   
Take a look please

File: src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp [hunk 1]
```diff
@@ -45,6 +46,9 @@ void SubgraphBaseTest::run() {
     summary.setDeviceName(targetDevice);
     summary.updateOPsStats(function, status);
 
+    if (targetDevice == "MYRIAD") {
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:49]: Why do you make such changes? I strongly disagree

**Review by iefode — CHANGES_REQUESTED** [review]

File: src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp (hunk 1 above)
Comment (KorA6) [reply src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp]: Changed.

File: src/tests/functional/shared_test_classes/src/single_layer/experimental_detectron_roifeatureextractor.cpp [hunk 2]
```diff
@@ -46,6 +47,9 @@ std::string ExperimentalDetectronROIFeatureExtractorLayerTest::getTestCaseName(
 }
 
 void ExperimentalDetectronROIFeatureExtractorLayerTest::SetUp() {
+    // w/a for myriad (cann't store 2 caches simultaneously)
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/single_layer/experimental_detectron_roifeatureextractor.cpp:50]: Minor: Could you please add the ticket number?>

Comment (KorA6) [reply src/tests/functional/shared_test_classes/src/single_layer/experimental_detectron_roifeatureextractor.cpp:51]: Done

**Pull request merged**